src/pages/events/custom-events.md

```diff
@@ -0,0 +1,123 @@
+---
+
+# Adding custom event
+
+In some cases, the client may need only events that meet certain conditions.
+
+For example, the client wants only to receive events when the stock of some products is low, currently, it's only
+possible to achieve this by sending all events on product or inventory updates and filtering them on the client side.
+Such an approach leads to sending a large amount of not necessary data and adding additional logic to filter it.
+
+You can subscribe either to available Commerce events or declare your own events as an extension to original event. In a
+case, if no subscription to original events was defined - only custom event will be emitted.
+
+## Defining custom event
+
+To create a custom event, it has to be defined in `io_events.xml` configuration.
+
+In the following example, we define a new event
+called `plugin.magento.catalog.model.resource_model.product.save_low_stock_event` when original
+event `plugin.magento.catalog.model.resource_model.product.save` is triggered.
+
+The required criteria for the event is when `qty` fields of original event is `less` than `20`
+
+```xml
+
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="...">
+    <event name="plugin.magento.catalog.model.resource_model.product.save_low_stock_event"
+           original="plugin.magento.catalog.model.resource_model.product.save">
+        <fields>
+            <field name="id"/>
+            <field name="sku"/>
+            <field name="qty"/>
+        </fields>
+        <rules>
+            <AndRule>
+                <field>qty</field>
+                <operator>lessThan</operator>
+                <value>20</value>
+            </AndRule>
+        </rules>
+    </event>
+</config>
+```
+
+Next configuration is required for custom event:
+
+| Field    | Description                                               |
+|----------|-----------------------------------------------------------|
+| name     | A unique name of event                                    |
+| original | An original event that has to be triggered                |
+| fields   | A set of fields from original event                       |
+| rules    | A set of rules that has to be met to trigger custom event |
+
+## Rules
+
+Rules can be defined as set or logical operators.
+
+| Rule    | Description      |
+|---------|------------------|
+| AndRule | Logical AND rule |
+| OrRule  | Logical OR rule  |
+
+A combination of rules can be used together.
+
+```xml
+
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="...">
+    <event name="plugin.magento.catalog.model.resource_model.product.save_low_stock_event"
+           from="plugin.magento.catalog.model.resource_model.product.save"
+    >
+        <fields>
+            <field name="id"/>
+            <field name="sku"/>
+            <field name="qty"/>
+        </fields>
+        <rules>
+            <AndRule>
```

Comment:
At the moment instead of <AndRule> and <OrRule> it's just implemented using `<rule>`:

```
        <rules>
            <rule>
                <field>qty</field>
                <operator>lessThan</operator>
                <value>20</value>
            </rule>
           ...
        </rules>    
```

Reply:
Thanks, fixed